### ISSUE TITLE
Warning that "parameterAddresses property of CustomFunction Invocation object requires manually created metadata" appears to no longer apply

### ISSUE THREAD

**Issue by chucklay** [OfficeDev/office-js-docs-pr#2472]
In the "Detect the address of a parameter" section of this page, there's a blue warning box that states that "The parameterAddresses property currently only works with manually-created JSON metadata." However, it appears that the default build script successfully handles this metadata, and creates a valid JSON metadata file, provided the JSDOC decorators for the function are correct.

It's probably worth keeping the rest of that notice for reference, but it seems like the first sentence no longer applies.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17035fc0-ca69-03f2-fe37-8189ff870cc7
* Version Independent ID: 2d43e719-043b-7a62-04bb-66b2cce7bf1d
* Content: [Options for Excel custom functions - Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-parameter-options?view=excel-js-preview&tabs=typescript)
* Content Source: [docs/excel/custom-functions-parameter-options.md](https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/excel/custom-functions-parameter-options.md)
* Product: **excel**
* Technology: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by AlexJerabek** [comment]
Thanks for bringing this potentially obsolete note to our attention. @alison-mk, can you please verify this?

**Comment by alison-mk** [comment]
Hi @chucklay, thank you! You're right, the issue that inspired this manually-created JSON note was very recently resolved. I've updated the article to incorporate your feedback. 
-Alison

**Comment by alison-mk** [comment]
Hi @chucklay, I'm following up on the "Detect the address of a parameter" documentation issue you submitted a few weeks ago. I have received varying reports of the JSON building properly with the @requiresParameterAddresses JSDoc tag. To help with my research, could you provide the versions of both node and npm that you're using to successfully build this JSON? Many thanks! -Alison

**Comment by chucklay** [comment]
Hi @alison-mk - Sure thing
Node: `v12.16.2`
NPM: `6.14.4`


I'm using this in a pair of functions that mimic H/VLOOKUP, with an optional default parameter that's returned in the event of the lookup failing. The metadata for these functions winds up looking like this:

```

        {
            "description": "Functionally identical to VLOOKUP, with an optional default on error.",
            "helpUrl": "https://plaidcloud.com/docs/excel-quick-start",
            "id": "VLOOKUP",
            "name": "VLOOKUP",
            "options": {
                "requiresParameterAddresses": true
            },
            "parameters": [
                {
                    "description": "the value you want to look up",
                    "name": "lookup_value",
                    "type": "any"
                },
                {
                    "description": "Where you want to look for it",
                    "dimensionality": "matrix",
                    "name": "table_array",
                    "type": "any"
                },
                {
                    "description": "Which column to lookup from",
                    "name": "col_index_num",
                    "type": "number"
                },
                {
                    "description": "TRUE - aproximate match, FALSE - exact match",
                    "name": "range_lookup",
                    "optional": true,
                    "type": "boolean"
                },
                {
                    "description": "A default value to use if no values are found",
                    "name": "default_val",
                    "optional": true,
                    "type": "any"
                }
            ],
            "result": {}
        },
        {
            "description": "Functionally identical to HLOOKUP, with an optional default on error.",
            "helpUrl": "https://plaidcloud.com/docs/excel-quick-start",
            "id": "HLOOKUP",
            "name": "HLOOKUP",
            "options": {
                "requiresParameterAddresses": true
            },
            "parameters": [
                {
                    "description": "the value you want to look up",
                    "name": "lookup_value",
                    "type": "any"
                },
                {
                    "description": "Where you want to look for it",
                    "dimensionality": "matrix",
                    "name": "table_array",
                    "type": "any"
                },
                {
                    "description": "Which column to lookup from",
                    "name": "col_index_num",
                    "type": "number"
                },
                {
                    "description": "TRUE - aproximate match, FALSE - exact match",
                    "name": "range_lookup",
                    "optional": true,
                    "type": "boolean"
                },
                {
                    "description": "A default value to use if no values are found",
                    "name": "default_val",
                    "optional": true,
                    "type": "any"
                }
            ],
            "result": {}
        }
```

And this is the result of the standard build process - no modifications.

**Comment by alison-mk** [comment]
Hi @chucklay, thank you for providing this information! Much appreciated. We've resolved the issue that caused the varying success reports I mentioned. 

I made one update to the code sample in the [Detect the address of a parameter](https://docs.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-parameter-options?tabs=javascript#detect-the-address-of-a-parameter) article section to help resolve this issue, and the update may be relevant to you. Here's the gist of the update: To ensure that the `result` object in the JSON metadata autogenerates with the `dimensionality` property set to `matrix`, the JSDoc needs an @returns tag that indicates a matrix of values, such as `@returns {string[][]}`. 

I'll re-close this issue now, but if you have any questions feel free to let us know.

Cheers, 
Alison
